api/metric/api.go

```diff
@@ -234,3 +243,21 @@ type resourceOption resource.Resource
 func (r resourceOption) Apply(config *Config) {
 	config.Resource = resource.Resource(r)
 }
+
+// WithLibraryName applies provided library name.  This is meant for
+// use in `Provider` implementations that have not used
+// `WrapMeterImpl`.  Implementations built using `WrapMeterImpl` have
+// instrument descriptors through this package are taken care.
```

Comment:
The last sentence here reads weird. Not sure, did you mean something like this:

```suggestion
// instrument descriptors taken care of through this package.
```